### PR TITLE
Replace admin create user interaction with facade

### PR DESCRIPTION
**Pull request by IniZio**
ref DEV-1645

### CODE REVIEW

File: pkg/admin/facade/user.go [hunk 1]
```diff
@@ -69,38 +65,25 @@ func (f *UserFacade) SearchPage(
...
-			Kind:                     interactionintents.IntentAuthenticateKindSignup,
-			SuppressIDPSessionCookie: true,
-		},
-		&createUserInput{
-			identityDef: identityDef,
-			password:    password,
+func (f *UserFacade) Create(identityDef model.IdentityDef, password string) (userID string, err error) {
+	// NOTE: identityDef is assumed to be a login ID since portal only supports login ID
+	loginIDInput := identityDef.(*model.IdentityDefLoginID)
+	loginIDKeyType := apimodel.LoginIDKeyType(loginIDInput.Key)
```
Comment (louischan-oursky) [inline pkg/admin/facade/user.go:71]: This is incorrect. You cannot simply say key == type. You use `LoginIDConfig.GetKeyConfig`. If not found, you return ```api.NewInvariantViolated(
				"InvalidLoginIDKey",
				"invalid login ID key",
				nil,
			)``` (as in pkg/lib/interaction/nodes/use_identity_login_id.go)

File: pkg/lib/facade/coordinator.go [hunk 2]
```diff
@@ -417,6 +426,127 @@ func (c *Coordinator) AuthenticatorVerifyOneWithSpec(userID string, authenticato
...
+	for _, t := range authenticatorTypes {
+		switch t {
+		case model.AuthenticatorTypePassword:
+			if len(password) == 0 {
+				return nil, api.NewInvariantViolated(
+					"PasswordRequired",
+					"password is required",
+					nil,
+				)
+			}
```
Comment (louischan-oursky) [inline pkg/lib/facade/coordinator.go:493]: This is incorrect. Password is not required when we can create other authenticator.

File: pkg/lib/facade/coordinator.go [hunk 3]
```diff
@@ -417,6 +426,127 @@ func (c *Coordinator) AuthenticatorVerifyOneWithSpec(userID string, authenticato
...
+	}
+
+	// 3. Create primary authenticators
+	authenticatorInfos, err := c.createPrimaryAuthenticators(identityInfo, userID, password)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(authenticatorInfos) == 0 {
+		return nil, api.InvalidConfiguration.New("no primary authenticator can be created for identity")
```
Comment (louischan-oursky) [inline pkg/lib/facade/coordinator.go:462]: To preserve the original behavior, we need to return ```api.NewInvariantViolated(
					"PasswordRequired",
					"password is required",
					nil,
				)``` if the project allows password, but password was not given.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]
mark this Request changes

File: pkg/lib/facade/coordinator.go (hunk 3 above)
Comment (IniZio) [reply pkg/lib/facade/coordinator.go]: Updated 🙏

**Review by louischan-oursky — CHANGES_REQUESTED** [review]
Given this config

```
authentication:
  identities:
  - login_id
  - oauth
  primary_authenticators:
  - oob_otp_email
  - password
```
and this request

```
mutation {
  createUser(input: {definition: {loginID: {value: "louischan+01@oursky.com", key: "email"}}}) {
    user {
      id
      standardAttributes
      identities {
        edges {
          node {
            id
            type
            claims
          }
        }
      }
      authenticators {
        edges {
          node {
            id
            type
            kind
            claims
          }
        }
      }
    }
  }
}
```

Gives this error

```
{
  "data": null,
  "errors": [
    {
      "message": "invalid target:\n<root>: format\n  map[format:phone]",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "createUser"
      ],
      "extensions": {
        "errorName": "Invalid",
        "info": {
          "causes": [
            {
              "location": "",
              "kind": "format",
              "details": {
                "format": "phone"
              }
            }
          ]
        },
        "reason": "ValidationFailed"
      }
    }
  ]
}
```

**Comment by IniZio** [comment]
> Given this config
> 
> ```
> authentication:
>   identities:
>   - login_id
>   - oauth
>   primary_authenticators:
>   - oob_otp_email
>   - password
> ```
> 
> and this request
> 
> ```
> mutation {
>   createUser(input: {definition: {loginID: {value: "louischan+01@oursky.com", key: "email"}}}) {
>     user {
>       id
>       standardAttributes
>       identities {
>         edges {
>           node {
>             id
>             type
>             claims
>           }
>         }
>       }
>       authenticators {
>         edges {
>           node {
>             id
>             type
>             kind
>             claims
>           }
>         }
>       }
>     }
>   }
> }
> ```
> 
> Gives this error
> 
> ```
> {
>   "data": null,
>   "errors": [
>     {
>       "message": "invalid target:\n<root>: format\n  map[format:phone]",
>       "locations": [
>         {
>           "line": 2,
>           "column": 3
>         }
>       ],
>       "path": [
>         "createUser"
>       ],
>       "extensions": {
>         "errorName": "Invalid",
>         "info": {
>           "causes": [
>             {
>               "location": "",
>               "kind": "format",
>               "details": {
>                 "format": "phone"
>               }
>             }
>           ]
>         },
>         "reason": "ValidationFailed"
>       }
>     }
>   ]
> }
> ```

Fixed in [66ac59b](https://github.com/authgear/authgear-server/pull/4519/commits/66ac59b6200ec9ccdb71a4b37dcab3886444b40c) :adore2x:

**Pull request merged**